wasm/jit/jit_arm64.go

```diff
@@ -59,35 +59,84 @@ func newCompiler(f *wasm.FunctionInstance, ir *wazeroir.CompilationResult) (comp
 		locationStack: newValueLocationStack(),
 	}
 
+	if ir != nil {
+		compiler.labels = make(map[string]*labelInfo, len(ir.LabelCallers))
+		for key, callers := range ir.LabelCallers {
+			compiler.labels[key] = &labelInfo{callers: callers}
+		}
+	}
 	return compiler, nil
 }
 
 type arm64Compiler struct {
 	builder *asm.Builder
 	f       *wasm.FunctionInstance
+	// setJmpTargetOnNextInstructions holds jmp kind instructions where we want to set the next coming
+	// instruction as the destination of these jmp instructions.
+	setJmpTargetOnNextInstructions []*obj.Prog
 	// locationStack holds the state of wazeroir virtual stack.
 	// and each item is either placed in register or the actual memory stack.
 	locationStack *valueLocationStack
+	// labels hold per wazeroir label specific information in this function.
+	labels map[string]*labelInfo
+	// maxStackPointer tracks the maximum value of stack pointer (from valueLocationStack).
+	maxStackPointer uint64
 }
 
 // compile implements compiler.compile for the arm64 architecture.
 func (c *arm64Compiler) compile() (code []byte, staticData compiledFunctionStaticData, maxStackPointer uint64, err error) {
+	// c.maxStackPointer tracks the maximum stack pointer across all valueLocationStack(s)
+	// used for all labels (via replaceLocationStack), excluding the current one.
+	// Hence, we check here if the final block's max one exceeds the current c.maxStackPointer.
+	maxStackPointer = c.maxStackPointer
+	if maxStackPointer < c.locationStack.maxStackPointer {
+		maxStackPointer = c.locationStack.maxStackPointer
+	}
+
 	code, err = mmapCodeSegment(c.builder.Assemble())
 	if err != nil {
 		return
 	}
 	return
 }
 
+type labelInfo struct {
+	// callers is the number of call sites which may jump into this label.
```

Comment:
"may" or is will this crash? wondering again if this is max vs ceil (ex call sites we are aware of due to compilation). This also clarifies impact, ex what is this for? if a max we crash on breach. if a ceil, we optimize according to it. Is it potentially zero? if so why would we care?

Reply:
no, may means we have conditional jumps (br_if) where we may or may not jump.

Reply:
Will clarify more in the comment.

Reply:
ok maybe if that's the case, is it ok to be literal and use the symbol name? 

Ex ... `brIfCount is the count of "br_if" (OpcodeBrIf) sites that may or may not jump to this label`

Where possible we can use symbol names in wasm to differentiate wasm lingo from arm64 ones, where using most precise wins

Reply:
I would just drop "may"! !https://github.com/tetratelabs/wazero/commit/ffcee6cd4aaa550862333dfd2b13a1f8e6f6a6fd